dev-vcs/pijul/pijul-1.0.0_beta4.ebuild

```diff
@@ -0,0 +1,386 @@
+	serde-1.0.152
+	serde_bytes-0.11.9
+	serde_derive-1.0.152
+	serde_json-1.0.93
+	serde_repr-0.1.10
+	serde_urlencoded-0.7.1
+	sha2-0.9.9
+	shell-words-1.1.0
+	signal-hook-registry-1.4.1
+	signature-1.6.4
+	slab-0.4.8
+	smallvec-1.10.0
+	socket2-0.4.7
+	static_assertions-1.1.0
+	strsim-0.10.0
+	subtle-2.4.1
+	syn-1.0.109
+	synstructure-0.12.6
+	tar-0.4.38
+	tempfile-3.4.0
+	termcolor-1.2.0
+	terminal_size-0.1.17
+	textwrap-0.16.0
+	thiserror-1.0.38
+	thiserror-impl-1.0.38
+	thread_local-1.1.7
+	threadpool-1.8.1
+	thrussh-0.33.5
+	thrussh-config-0.5.0
+	thrussh-keys-0.21.0
+	thrussh-libsodium-0.2.1
+	time-0.1.45
+	tint-1.0.1
+	tinyvec-1.6.0
+	tinyvec_macros-0.1.1
+	tokio-1.25.0
+	tokio-macros-1.8.2
+	tokio-native-tls-0.3.1
+	tokio-stream-0.1.12
+	tokio-util-0.7.7
+	toml-0.5.11
+	toml_datetime-0.6.1
+	toml_edit-0.19.4
+	tower-service-0.3.2
+	tracing-0.1.37
+	tracing-core-0.1.30
+	try-lock-0.2.4
+	twox-hash-1.6.3
+	typenum-1.16.0
+	unicode-bidi-0.3.10
+	unicode-ident-1.0.6
+	unicode-normalization-0.1.22
+	unicode-width-0.1.10
+	unicode-xid-0.2.4
+	url-2.3.1
+	uuid-0.8.2
+	validator-0.15.0
+	vcpkg-0.2.15
+	version_check-0.9.4
+	waker-fn-1.1.0
+	walkdir-2.3.2
+	want-0.3.0
+	wasi-0.10.0+wasi-snapshot-preview1
+	wasi-0.11.0+wasi-snapshot-preview1
+	wasi-0.9.0+wasi-snapshot-preview1
+	wasm-bindgen-0.2.84
+	wasm-bindgen-backend-0.2.84
+	wasm-bindgen-futures-0.4.34
+	wasm-bindgen-macro-0.2.84
+	wasm-bindgen-macro-support-0.2.84
+	wasm-bindgen-shared-0.2.84
+	wasm-streams-0.2.3
+	web-sys-0.3.61
+	wepoll-ffi-0.1.2
+	which-4.4.0
+	whoami-1.3.0
+	winapi-0.3.9
+	winapi-i686-pc-windows-gnu-0.4.0
+	winapi-util-0.1.5
+	winapi-x86_64-pc-windows-gnu-0.4.0
+	windows-sys-0.42.0
+	windows-sys-0.45.0
+	windows-targets-0.42.1
+	windows_aarch64_gnullvm-0.42.1
+	windows_aarch64_msvc-0.42.1
+	windows_i686_gnu-0.42.1
+	windows_i686_msvc-0.42.1
+	windows_x86_64_gnu-0.42.1
+	windows_x86_64_gnullvm-0.42.1
+	windows_x86_64_msvc-0.42.1
+	winnow-0.3.3
+	winreg-0.10.1
+	xattr-0.2.3
+	yasna-0.4.0
+	zbus-1.9.3
+	zbus_macros-1.9.3
+	zeroize-1.3.0
+	zeroize_derive-1.3.3
+	zstd-seekable-0.1.14
+	zvariant-2.10.0
+	zvariant_derive-2.10.0
+	${PN}-${MY_PIJUL_PV}
+"
+
+inherit cargo
+
+DESCRIPTION="A distributed version control system."
+HOMEPAGE="https://pijul.org/ https://nest.pijul.com/pijul/pijul"
+SRC_URI="
+	$(cargo_crate_uris)
+"
+RESTRICT="mirror"
+
+LICENSE="GPL-2+"
+# Dependent crate licenses
+LICENSE+="
+	Apache-2.0 BSD-2 BSD CC0-1.0 GPL-2 GPL-2+ MIT Unicode-DFS-2016 ZLIB
+"
+SLOT="0"
+KEYWORDS="~amd64"
+S="${WORKDIR}/${PN}-${MY_PIJUL_PV}"
+
+IUSE="git"
+BDEPEND="virtual/rust"
+
+src_prepare() {
+	default
+
+	# We get the sources from one of the crates, as the Pijul source is managed
+	# with Pijul on nest.pijul.com - we obviously can't clone it (because we'd
+	# need a working pijul binary to do that), and the Nest doesn't support
+	# fetching archives over HTTP or anything like that.
+	cp -af "${WORKDIR}/cargo_home/gentoo/${PN}-${MY_PIJUL_PV}" "${WORKDIR}" || die
```

Comment:
I guess I could also set `S` appropriately and not override src_prepare() at all...

Reply:
done.